Zend/zend_alloc.c

```diff
@@ -1914,7 +1914,7 @@ static zend_mm_heap *zend_mm_init(void)
 	heap->peak = 0;
 #endif
 #if ZEND_MM_LIMIT
-	heap->limit = ((size_t)Z_L(-1) >> (size_t)Z_L(1));
+	heap->limit = Z_UL(-1) >> Z_UL(1);
```

Comment:
```suggestion
	heap->limit = Z_UL(-1) >> 1;
```

the type should be determined by the left operand at least when the right operand is constant